src/reducers/provider-state-updaters.js

```diff
@@ -0,0 +1,200 @@
+// THE SOFTWARE.
+
+import {withTask} from 'react-palm/tasks';
+import {default as Console} from 'global/console';
+import {generateHashId, getError} from 'utils/utils';
+import {EXPORT_FILE_TO_CLOUD_TASK, ACTION_TASK, DELAY_TASK} from 'tasks/tasks';
+import {exportFileSuccess, exportFileError, saveToCloudSuccess} from 'actions/provider-actions';
+import {
+  removeNotification,
+  toggleModal,
+  addNotification
+} from 'actions/ui-state-actions';
+import {
+  DEFAULT_NOTIFICATION_TYPES,
+  DEFAULT_NOTIFICATION_TOPICS
+} from 'constants/default-settings';
+
+export const INITIAL_PROVIDER_STATE = {
+  isLoading: false,
+  error: null,
+  currentProvider: null,
+  successInfo: {}
+};
+
+function createFileJson(mapData) {
+  const {map, info} = mapData;
+  const data = JSON.stringify(map);
+  const newBlob = new Blob([data], {type: 'application/json'});
+
+  // TODO: Allow user import file name
+  const name =
+    info.title && info.title.length
+      ? info.title
+      : `keplergl_${generateHashId(6)}`;
+  const fileName = `${name}.json`;
+
+  return {file: new File([newBlob], fileName), fileName};
+}
+
+function createActionTask(action, payload) {
+  if (typeof action === 'function') {
+    return ACTION_TASK().map(_ => action(payload));
+  }
+
+  return null;
+}
+
+/**
+ * This method will export the current kepler config file to the chosen cloud proder
+ * add returns a share URL
+ *
+ * @param {*} state
+ * @param {*} action
+ */
+export const exportFileToCloudUpdater = (state, action) => {
+  const {mapData, provider, isPublic, onSuccess, onError, closeModal} = action.payload;
+  console.log(mapData);
+
+  if (!provider) {
+    Console.error(`provider is not defined`);
+    return state;
+  }
+
+  if (typeof provider.uploadFile !== 'function') {
+    Console.error(
+      `uploadFile is not a function of Cloud provider: ${provider.name}`
+    );
+    return state;
+  }
+
+  // TODO: do we need to always create a fileBlob?
+  // if this is provider specific, we should just send over the json
+  // and let provider decide what format to use
+  const {file, fileName} = createFileJson(mapData);
+
+  const newState = {
+    ...state,
+    isLoading: true,
+    currentProvider: provider.name
+  };
+
+  // payload called by provider.uploadFile
+  const payload = {
+    mapData,
+    blob: file,
+    fileName,
+    isPublic
+  };
+
+  const uploadFileTask = EXPORT_FILE_TO_CLOUD_TASK({provider, payload}).bimap(
+    // success
+    response => exportFileSuccess({response, provider, onSuccess, closeModal}),
+    // error
+    error => exportFileError({error, provider, onError})
+  );
+
+  return withTask(newState, uploadFileTask);
+};
+
+/**
+ *
+ * @param {*} state
+ * @param {*} action
+ */
+export const exportFileSuccessUpdater = (state, action) => {
+  const {response, provider, onSuccess, closeModal} = action.payload;
+
+  const newState = {
+    ...state,
+    isLoading: false,
+    // TODO: do we always have to store this?
```

Comment:
i think we should store this since each provider may want to display information upon a successful operation. Since data may be dynamic we may have to define a set of standard properties each provider should provider upon operation completion